internal/config/config.go

```diff
@@ -56,79 +53,31 @@ func LoadMap(m map[string]interface{}) error {
 }
 
 func doLoad(sources ...config.YAMLOption) error {
-	opts := append(sources, config.Expand(os.LookupEnv)) //nolint:gocritic
-	provider, err := config.NewYAML(opts...)
+	provider, err := mkProvider(sources...)
 	if err != nil {
-		if strings.Contains(err.Error(), "couldn't expand environment") {
-			return fmt.Errorf("error loading configuration due to unknown environment variable. Config values containing '$' are interpreted as environment variables. Use '$$' to escape literal '$' values: [%w]", err)
-		}
-		return fmt.Errorf("failed to load config: %w", err)
+		return err
 	}
 
 	conf.replaceProvider(provider)
-
 	return nil
 }
 
-// LoadAndWatch automatically reloads configuration if the config file changes.
-func LoadAndWatch(ctx context.Context, confFile string, overrides map[string]interface{}) error {
```

Comment:
This was dead code?

Reply:
Yes. It was added very early on but I never followed through with it. Now it doesn't make sense anymore because most components only read the configuration only once on startup and can't be reloaded easily.